tidy3d/components/tcad/simulation/heat_charge.py

```diff
@@ -1778,3 +1786,129 @@ def _useHeatSourceFromConductionSim(self):
+                                            new_boxes.append(doping_box)
+                                        sc_medium = sc_medium.updated_copy(**{key: new_boxes})
+                            struct_medium[structure] = structure.medium.updated_copy(
+                                charge=sc_medium
+                            )
+                        else:
+                            struct_medium[structure] = structure.medium
+
+                    # change structures
+                    new_structures = []
+                    for structure in self.structures:
+                        new_center = list(structure.geometry.center)
+                        new_center[zero_dims[0]] = structure.geometry.center[2]
+                        new_center[2] = structure.geometry.center[zero_dims[0]]
+
+                        new_size = list(structure.geometry.size)
+                        new_size[zero_dims[0]] = structure.geometry.size[2]
+                        new_size[2] = structure.geometry.size[zero_dims[0]]
+                        new_structures.append(
+                            structure.updated_copy(
+                                geometry=structure.geometry.updated_copy(
+                                    center=new_center, size=new_size
+                                ),
+                                medium=struct_medium[structure],
+                            )
+                        )
+
+                    # Boundary conditions
+                    new_boundary_spec = []
+                    for boundary in self.boundary_spec:
+                        new_placement = boundary.placement
+                        PlacementTypes = (SimulationBoundary, StructureSimulationBoundary)
+                        if isinstance(boundary.placement, PlacementTypes):
+                            surfaces = []
+                            for s in boundary.placement.surfaces:
+                                if "z" in s:
+                                    new_s = s.replace("z", ["x", "y"][zero_dims[0]])
+                                    surfaces.append(new_s)
+                                else:
+                                    surfaces.append(s)
+                            new_placement = new_placement.updated_copy(surfaces=surfaces)
+
+                        new_boundary_spec.append(boundary.updated_copy(placement=new_placement))
+
+                    # Monitors
+                    new_monitors = []
+                    for mnt in self.monitors:
+                        new_center = list(mnt.center)
+                        new_center[zero_dims[0]] = mnt.center[2]
+                        new_center[2] = mnt.center[zero_dims[0]]
+
+                        new_size = list(mnt.size)
+                        new_size[zero_dims[0]] = mnt.size[2]
+                        new_size[2] = mnt.size[zero_dims[0]]
+
+                        new_monitors.append(mnt.updated_copy(center=new_center, size=new_size))
```

Comment:
looks like this kind of operation is performed 3-4 times, I think it's worth putting it in a function something like:
```
doping_box_tmp = rotated_copy(doping_box)
new_boxes.append(doping_box_tmp.updated_copy(source=source))
...
new_structures.append(
    structure.updated_copy(
        geometry=rotated_copy(structure.geometry),
        medium=struct_medium[structure],
    )
)
...
new_monitors.append(rotated_copy(mnt))
...
```